src/Node/SetNode.php

```diff
@@ -33,9 +33,15 @@ public function __construct(bool $capture, Node $names, Node $values, int $linen
         $safe = false;
         if ($capture) {
             $safe = true;
-            if ($values instanceof TextNode) {
+            if (Node::class === get_class($values) && !count($values)) {
+                $values = new ConstantExpression('', $values->getTemplateLine());
+                $capture = false;
+            } elseif ($values instanceof TextNode) {
                 $values = new ConstantExpression($values->getAttribute('data'), $values->getTemplateLine());
                 $capture = false;
+            } elseif ($values instanceof PrintNode && $values->getNode('expr') instanceof ConstantExpression) {
+                $values = new ConstantExpression($values->getNode('expr')->getAttribute('value'), $values->getTemplateLine());
```

Comment:
why creating a new ConstantExpression here instead of using `$values->getNode('expr')` directly ?